habitat_baselines/common/base_trainer.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC, abstractmethod
+
+from habitat_baselines.common.trainer_registry import trainer_registry
+
+
+class BaseTrainer(ABC):
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def train(self):
+        pass
+
+    @abstractmethod
+    def eval(self):
+        pass
+
+    @abstractmethod
+    def save_checkpoint(self, file_name):
+        pass
+
+    @abstractmethod
+    def load_checkpoint(self):
+        pass
+
+
+def get_trainer(trainer_name, trainer_cfg):
```

Comment:
At this point, i will suggest to move all of the `make_*` and `get_*` functions to a separate file in utils. They are all over the place now.